01-func.js

```diff
@@ -0,0 +1,150 @@
+"use strict";
+
+/**
+ * @typedef {Object} City
+ * @property {string} city
+ * @property {number} population
+ * @property {number} area
+ * @property {number} density
+ * @property {string} country
+ * @property {number} normalizedDensity
+ */
+
+/**
+ * @param {string} data
+ * @param {boolean} removeHeaderLine
+ * @param {string} divider
+ * @returns {Array<Array<string>>}
+ */
+const parseDataToArray = (
+  str,
+  removeHeaderLine = true,
+  endOfRow = "\n",
+  divider = ",",
+) => {
+  const parsedData = str.split(endOfRow).map((line) => line.split(divider));
+  if (removeHeaderLine) parsedData.shift();
+  return parsedData;
+};
+
+/**
+ * @param {Array<Array<string>>} arr
+ * @param {function} filterFn
+ * @param {function} buildFn
+ * @returns {Array<Array>}
+ * */
+const filterAndBuildData = (arr, filterFn = (el) => el, buildFn) => {
```

Comment:
function has two responsibilities (build and filter). Maybe split it to correspond SRP